packages/dds/map/src/directory.ts

```diff
@@ -1475,8 +1482,23 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
      * @param op - The message if from a remote delete, or null if from a local delete
      */
     private deleteSubDirectoryCore(subdirName: string, local: boolean) {
-        // This should make the subdirectory structure unreachable so it can be GC'd and won't appear in snapshots
-        // Might want to consider cleaning out the structure more exhaustively though?
-        return this._subdirectories.delete(subdirName);
+        if (this.hasSubDirectory(subdirName)) {
+            const previousValue = this.getSubDirectory(subdirName);
+            assert(previousValue !== undefined, "Subdirectory should be expected");
+            // This should make the subdirectory structure unreachable so it can be GC'd and won't appear in snapshots
+            // Might want to consider cleaning out the structure more exhaustively though?
+            const successfullyRemoved = this._subdirectories.delete(subdirName);
+            if (successfullyRemoved) {
+                const event: ISubDirectoryDeleted = {
+                    key: subdirName,
+                    path: this.absolutePath,
+                    previousValue,
+                };
+                this.directory.emit("subDirectoryDeleted", event, local, this.directory);
```

Comment:
I'd think you want to provide a path for sub-directory being deleted (to be similar to creation event). And I think we have to have a discussion (and separate bug) on what happens to sub-directory object after that (as various entities can hold to it). I'd think that we want
1. Make it disposed, such that any call on this object fails.
2. Have an event on sub-directory itself, that it's gone
3. Not have it on this event.
If that's the direction, then I'd argue we open a bug for # 1-2, and make # 3 in this PR.

Reply:
Need to open an issue if we are pushing the work to future in this space